thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/metric/transfer/MetricTransfer.java

```diff
@@ -31,20 +43,52 @@ public static void rescaleMetric(MetricTimeSeries metricsToModify, String metric
       scaledValues = new ArrayList<>();
     }
 
-    // Went over the List of Scaling Factors and Rescale the timestamps with in the time range
-    // get the metric name
-    for (long ts : metricsToModify.getTimeWindowSet()) {
+    int seasonalSize = Integer.parseInt(properties.getProperty(SEASONAL_SIZE, DEFAULT_SEASONAL_SIZE));
+    TimeUnit seasonalUnit = TimeUnit.valueOf(properties.getProperty(SEASONAL_UNIT, DEFAULT_SEASONAL_UNIT));
+    long seasonalMillis = seasonalUnit.toMillis(seasonalSize);
+    int baselineSeasonalPeriod = Integer.parseInt(properties.getProperty(BASELINE_SEASONAL_PERIOD, DEFAULT_BASELINE_SEASONAL_PERIOD));
+
+    // The scaling model works as follows:
+    // 1. If the time range of scaling factor overlaps with current time series, then the scaling
+    //    factor is applied to all corresponding baseline time series.
+    // 2. If the time range of scaling factor overlaps with a baseline time series, then baseline
+    //    value that overlaps with the scaling factor is set to 0. Note that in current
+    //    implementation of ThirdEye value 0 means missing data. Thus, the baseline value will be
+    //    removed from baseline calculation.
+    //
+    // The model is implemented as follows:
+    // 1. Check if a timestamp located in any time range of scaling factor.
+    // 2. If it is, then check if it is a current value or baseline value.
+    // 3a. If it is a baseline value, then set its value to 0.
+    // 3b. If it is a current value, then apply the scaling factor to its corresponding baseline
+    //     values.
+    Set<Long> timeWindowSet = metricsToModify.getTimeWindowSet();
+    for (long ts : timeWindowSet) {
       for (ScalingFactor sf: scalingFactorList) {
         if (sf.isInTimeWindow(ts)) {
-          double originalValue = metricsToModify.get(ts, metricName).doubleValue();
-          double scaledValue = originalValue * sf.getScalingFactor();
-          metricsToModify.set(ts, metricName, scaledValue);
-          if (DEBUG) {
-            scaledValues.add(new ScaledValue(ts, originalValue, scaledValue));
+          if (ts < windowStartTime) { // the timestamp belongs to a baseline time series
+            if (DEBUG) {
+              double originalValue = metricsToModify.get(ts, metricName).doubleValue();
+              scaledValues.add(new ScaledValue(ts, originalValue, 0.0));
+            }
+            metricsToModify.set(ts, metricName, 0.0); // zero will be removed in analyze function
```

Comment:
Removing a TS from metricToModify is basically a hack under the current design of ThirdEye. Since we already have a hack which removes TS, I don't want to create another hack here. 

I would suggest to solve this problem once ThirdEye is able to distinguish value 0 and an empty value. Otherwise, we might lose track of all the hacks we did for experimental use cases.